#!/usr/bin/env python
PACKAGE = "slave_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kpm_x", double_t, 0, "-x Master Pos Gain", 3.0, 0.0, 200)
gen.add("Kpm_y", double_t, 0, "-y Master Pos Gain", 3.0, 0.0, 200)
gen.add("Kpm_z", double_t, 0, "-z Master Pos Gain", 5.0, 0.0, 200)

gen.add("Kdm_x", double_t, 0, "-x Master Damp Gain", 0.1, 0.0, 10)
gen.add("Kdm_y", double_t, 0, "-y Master Damp Gain", 0.1, 0.0, 10)
gen.add("Kdm_z", double_t, 0, "-z Master Damp Gain", 0.2, 0.0, 10)

gen.add("Kfm_x", double_t, 0, "-x Master Force Gain", 0.30, 0.0, 2.0)
gen.add("Kfm_y", double_t, 0, "-y Master Force Gain", 0.30, 0.0, 2.0)
gen.add("Kfm_z", double_t, 0, "-z Master Force Gain", 0.20, 0.0, 2.0)

gen.add("Kps_x", double_t, 0, "-x Slave Pos Gain", 8.0, 0.0, 200)
gen.add("Kps_y", double_t, 0, "-y Slave Pos Gain", 8.0, 0.0, 200)
gen.add("Kps_z", double_t, 0, "-z Slave Pos Gain", 8.0, 0.0, 200)

gen.add("Kds_x", double_t, 0, "-x Slave Damp Gain", 1.0, 0.0, 10)
gen.add("Kds_y", double_t, 0, "-y Slave Damp Gain", 1.0, 0.0, 10)
gen.add("Kds_z", double_t, 0, "-z Slave Damp Gain", 1.0, 0.0, 10)

gen.add("Kfs_x", double_t, 0, "-x Slave Force Gain", 0.0, 0.0, 2.0)
gen.add("Kfs_y", double_t, 0, "-y Slave Force Gain", 0.0, 0.0, 2.0)
gen.add("Kfs_z", double_t, 0, "-z Slave Force Gain", 0.0, 0.0, 2.0)

gen.add("b_x", double_t, 0, "-x Wave Impedance", 0.95, 0.0, 3.0)
gen.add("b_y", double_t, 0, "-y Wave Impedance", 1.0, 0.0, 3.0)
gen.add("b_z", double_t, 0, "-z Wave Impedance", 1.1, 0.0, 3.0)

gen.add("CSel", int_t, 0, "Controller selector: 0 = 4C, 1 = WV, 2 = PC, 3 = reset", 0, 0, 3)

gen.add("WVfilt_x", int_t, 0, "-x Wave variable filter cut-off frequency [Hz]", 30, 1, 1000)
gen.add("WVfilt_y", int_t, 0, "-y Wave variable filter cut-off frequency [Hz]", 30, 1, 1000)
gen.add("WVfilt_z", int_t, 0, "-z Wave variable filter cut-off frequency [Hz]", 25, 1, 1000)

gen.add("PClim_x", double_t, 0, "-x Passivity control dissipation force limiter", 2.5, 1, 1000)
gen.add("PClim_y", double_t, 0, "-y Passivity control dissipation force limiter", 2.5, 1, 1000)
gen.add("PClim_z", double_t, 0, "-z Passivity control dissipation force limiter", 3.0, 1, 1000)

gen.add("velfilt_m_x", double_t, 0, "-x Master velocity filter cut-off frequency [Hz]", 100, 1, 1000)
gen.add("velfilt_m_y", double_t, 0, "-y Master velocity filter cut-off frequency [Hz]", 100, 1, 1000)
gen.add("velfilt_m_z", double_t, 0, "-z Master velocity filter cut-off frequency [Hz]", 100, 1, 1000)

gen.add("velfilt_s_x", double_t, 0, "-x Slave velocity filter cut-off frequency [Hz]", 100, 1, 1000)
gen.add("velfilt_s_y", double_t, 0, "-y Slave velocity filter cut-off frequency [Hz]", 100, 1, 1000)
gen.add("velfilt_s_z", double_t, 0, "-z Slave velocity filter cut-off frequency [Hz]", 100, 1, 1000)

gen.add("forcefilt_e_x", double_t, 0, "-x Environment force filter cut-off frequency [Hz]", 100, 1, 1000)
gen.add("forcefilt_e_y", double_t, 0, "-y Environment force filter cut-off frequency [Hz]", 100, 1, 1000)
gen.add("forcefilt_e_z", double_t, 0, "-z Environment force filter cut-off frequency [Hz]", 50, 1, 1000)

gen.add("forcefilt_h_x", double_t, 0, "-x Human force filter cut-off frequency [Hz]", 50, 1, 1000)
gen.add("forcefilt_h_y", double_t, 0, "-y Human force filter cut-off frequency [Hz]", 50, 1, 1000)
gen.add("forcefilt_h_z", double_t, 0, "-z Human force filter cut-off frequency [Hz]", 50, 1, 1000)

gen.add("forcefilt_m_x", double_t, 0, "-x Master controller force filter cut-off frequency [Hz]", 35, 1, 1000)
gen.add("forcefilt_m_y", double_t, 0, "-y Master controller force filter cut-off frequency [Hz]", 30, 1, 1000)
gen.add("forcefilt_m_z", double_t, 0, "-z Master controller force filter cut-off frequency [Hz]", 25, 1, 1000)

gen.add("forcefilt_s_x", double_t, 0, "-x Slave controller force filter cut-off frequency [Hz]", 10, 1, 1000)
gen.add("forcefilt_s_y", double_t, 0, "-y Slave controller force filter cut-off frequency [Hz]", 10, 1, 1000)
gen.add("forcefilt_s_z", double_t, 0, "-z Slave controller force filter cut-off frequency [Hz]", 10, 1, 1000)

gen.add("posloopfilt_s_x", double_t, 0, "-x Slave controller position sync loop filter cut-off frequency [Hz]", 5, 1, 1000)
gen.add("posloopfilt_s_y", double_t, 0, "-y Slave controller position sync loop filter cut-off frequency [Hz]", 5, 1, 1000)
gen.add("posloopfilt_s_z", double_t, 0, "-z Slave controller position sync loop filter cut-off frequency [Hz]", 5, 1, 1000)


exit(gen.generate(PACKAGE, "slave_control", "parameters"))